schema_test.go

```diff
@@ -0,0 +1,121 @@
+
+	// mixed types
+	err = validatePublicSettings(`{"commandToExecute": "date", "fileUris":["https://a.b/c.txt?d=e&f=g", 0]}`)
+	require.NotNil(t, err)
+	require.Contains(t, err.Error(), "Expected: string, given: integer")
+}
+
+func TestValidatePublicSettings_timestampSupported(t *testing.T) {
+	require.Nil(t, validatePublicSettings(`{"commandToExecute": "date", "timestamp": 1}`))
+}
+
+func TestValidateProtectedSettings_empty(t *testing.T) {
+	require.Nil(t, validateProtectedSettings(""), "empty string")
+	require.Nil(t, validateProtectedSettings("{}"), "empty string")
+}
+
+func TestValidateProtectedSettings_unrecognizedField(t *testing.T) {
+	err := validateProtectedSettings(`{"alien":0}`)
+	require.NotNil(t, err)
+	require.Contains(t, err.Error(), "Additional property alien is not allowed")
+}
+
+func TestValidateProtectedSettings_commandToExecute(t *testing.T) {
+	// Invalid type
+	err := validateProtectedSettings(`{"commandToExecute": false}`)
+	require.NotNil(t, err)
+	require.Contains(t, err.Error(), "Expected: string, given: boolean")
+
+	// Valid
+	require.Nil(t, validateProtectedSettings(`{"commandToExecute":"date"}`))
+}
+
+func TestValidateProtectedSettings_storageAccountName(t *testing.T) {
+	chkPatternMismatch := func(e error, reason string) {
+		require.NotNil(t, e, reason)
+		require.Contains(t, e.Error(), "storageAccountName: Does not match pattern", reason)
+	}
+
+	// Specified but empty
+	chkPatternMismatch(validateProtectedSettings(`{"storageAccountName": ""}`), "empty")
+
+	// Too short
+	chkPatternMismatch(validateProtectedSettings(`{"storageAccountName": "aa"}`), "too short")
+
+	// Too long
+	chkPatternMismatch(validateProtectedSettings(`{"storageAccountName": "1234567890123456789012345"}`), "too long")
+
+	// invalid chars
+	chkPatternMismatch(validateProtectedSettings(`{"storageAccountName": "foo-bar"}`), "invalid char")
+	chkPatternMismatch(validateProtectedSettings(`{"storageAccountName": " foobar"}`), "invalid char")
+	chkPatternMismatch(validateProtectedSettings(`{"storageAccountName": "foobar "}`), "invalid char")
+	chkPatternMismatch(validateProtectedSettings(`{"storageAccountName": "foo.bar"}`), "invalid char")
+
+	// ok storage account name
+	require.Nil(t, validateProtectedSettings(`{"storageAccountName": "0foobarquuxyz12345"}`))
+}
+
+func TestValidateProtectedSettings_storageAccountKey(t *testing.T) {
+	chkPatternMismatch := func(e error, reason string) {
+		require.NotNil(t, e, reason)
+		require.Contains(t, e.Error(), "storageAccountKey: Does not match pattern", reason)
+	}
+
+	// empty
+	chkPatternMismatch(validateProtectedSettings(`{"storageAccountKey": ""}`), "empty")
+
+	// bad string
+	chkPatternMismatch(validateProtectedSettings(`{"storageAccountKey": "NotABase64Really!"}`), "not b64")
+
+	// missing '=' at the end
+	chkPatternMismatch(validateProtectedSettings(`{"storageAccountKey": "OllwYfXmC0mSMhWg4x+lUdLg6Eoa/d44+PxPTXBaadO5l87L4JzgkyyVvQr8r60WIzG2X8r6LLxkhNBQaHa3XQ="}`), "bad b64")
```

Comment:
or maybe we should keep this test because it actually validates that given a string of appropriate length, removing the `=` is caught by this test. If somebody changes the regex in the schema to a very simple regex that appears to be covering base64, this test will catch it!